### PR TITLE
Add support for tax_query operator EXISTS and NOT EXISTS

### LINKED ISSUE #893: tax_query with operator 'NOT EXIST' failed to translate
Is this a bug, question or feature request?
Bug

Describe the issue you encountered:
tax_query accept operator "NOT EXIST", but it failed to translate properly to ES query

WP_Query sample
```
new WP_Query(
		array(
			'ep_integrate'   => true,
			'post_type'     => array(
				'post'
			),
			'post_status'   => 'publish',
			'tax_query' => array(
				'relation' => 'OR',
				array(
					'taxonomy' => 'category',
					'field'    => 'slug',
					'terms'    => 'art-design',
					'operator' => 'IN'
				),
				array(
					'taxonomy' => 'category',
					'field'    => 'slug',
					'terms'    => 'uncategorized',
					'operator' => 'NOT EXISTS'
				),
			),
		)
	);
```

Translated to:
```
{
    "from": 0,
    "size": 10,
    "sort": [
        {
            "post_date": {
                "order": "desc"
            }
        }
    ],
    "query": {
        "match_all": {
            "boost": 1
        }
    },
    "post_filter": {
        "bool": {
            "must": [
                {
                    "bool": {
                        "should": [
                            {
                                "terms": {
                                    "terms.category.slug": [
                                        "art-design"
                                    ]
                                }
                            },
                            {
                                "terms": {
                                    "terms.category.slug": [
                                        "uncategorized"
                                    ]
                                }
                            }
                        ]
                    }
                },
                {
                    "term": {
                        "post_type.raw": "post"
                    }
                },
                {
                    "term": {
                        "post_status": "publish"
                    }
                }
            ]
        }
    }
}
```

### PR DESCRIPTION
**Pull request by ivankristianto**
fix #893

This PR is to add support for EXISTS and NOT EXISTS operator for tax_query.
Also added the unit testing

### CODE REVIEW

File: classes/class-ep-api.php [hunk 1]
```diff
@@ -1122,8 +1122,8 @@ public function format_args( $args ) {
 			$es_tax_query = array();
 
 			foreach( $args['tax_query'] as $single_tax_query ) {
-				if ( ! empty( $single_tax_query['terms'] ) ) {
-					$terms = (array) $single_tax_query['terms'];
+				if ( ! empty( $single_tax_query['taxonomy'] ) ) {
+					$terms = isset( $single_tax_query['terms'] )? (array) $single_tax_query['terms'] : array();
```
Comment (oscarssanchez) [inline classes/class-ep-api.php]: Hi @ivankristianto, maybe add a whitespace after `isset( $single_tax_query['terms'] )`

File: classes/class-ep-api.php [hunk 2]
```diff
@@ -1138,44 +1138,83 @@ public function format_args( $args ) {
...
-						 *
-						 * @since 2.4
-						 */
-
-						$and_nest = array(
-							'bool' => array(
-								'must' => array()
-							),
-						);
+					switch ( $operator ) {
```
Comment (oscarssanchez) [inline classes/class-ep-api.php]: This looks like a great use for the switch statement :+1:

**Comment by ivankristianto** [comment]
@oscarssanchez I have fixed the code based on your review.

**Comment by allan23** [comment]
Thanks @ivankristianto

**Pull request merged**